templates/bundles/TwigBundle/Exception/error.html.twig

```diff
@@ -0,0 +1,26 @@
+{% extends '@W3CWebsiteTemplates/base.html.twig' %}
+
+{% block main %}
+    <h1>{% block title %}{{ page.title|default('Error: {{ status_code }} - {{ status_text }}') }} | W3C{% endblock %}</h1>
+    <p>{{ 'error.intro'|trans({'status_code': status_code, 'status_text': status_text}, 'w3c_website_templates_bundle') }}</p>
+
+    {% block stacktrace %}
+        {% if is_granted('ROLE_ADMIN') and exception %}
+            <h2>Error details</h2>
+            <p>Message: <strong>{{ exception.message }}</strong></p>
+            <p>in <em>{{ exception.file }}: {{ exception.line }}</em></p>
+            <p>Trace:</p>
+            <pre>{{ exception.traceAsString }}</pre>
+        {% endif %}
+    {% endblock stacktrace %}
+
+    {% block content %}{% endblock content %}
+    <p>
+        {% set subject = "subject=Error: " ~ (exception ? exception.message : app.request.uri) %}
+        {% set body = "body=Issue description%0A%0A%0A%0A----%0AURL: " ~ app.request.uri ~ "%0ADate: " ~ "now"|date('c') %}
+        {% if exception %}
+            {% set body = body ~ "%0AException at " ~ exception.file ~ ": " ~ exception.line %}
+        {% endif %}
+        {{ 'error.contact'|trans({}, 'w3c_website_templates_bundle') }} <a href="mailto:sysreq@w3.org?{{ subject }}&{{ body }}">sysreq@w3.org</a>.
+    </p>
```

Comment:
For readability and security, I suggest this:
```suggestion
    <p>
        {% set subject -%}
            {%- apply url_encode -%}
                Error: {{ (exception ? exception.message : app.request.uri) }}
            {%- endapply -%}
        {%- endset %}
        {% set body -%}
            {%- apply url_encode -%}
Issue description



----
URL: {{ app.request.uri }}
Date: {{ "now"|date('c') }}
{% if exception %}
    Exception at {{ exception.file }}: {{ exception.line }}
{% endif %}
            {%- endapply -%}
        {%- endset %}
        {{ 'error.contact'|trans({}, 'w3c_website_templates_bundle')|raw }} <a href="mailto:sysreq@w3.org?subject={{ subject }}&body={{ body }}">sysreq@w3.org</a>.
    </p>
```
That way we are sure to encode everything (spaces need to be encoded for example)
Also apply "raw" on error.contact as it contains html.

Reply:
thanks for the suggestion. Integrated in https://github.com/w3c/w3c-website-templates-bundle/pull/40/commits/776d66e897902c8c85388521c3a2561fa634b319

I just noticed that added `exception.file` might provide too much details as it would display the complete path to the file. I'm not sure how harmful sharing that kind of information can be.

Reply:
> I just noticed that added exception.file might provide too much details as it would display the complete path to the file. I'm not sure how harmful sharing that kind of information can be.

Some ideas:
- we could restrict this info to admins only
- we could only display the filename and not the full path.

Based on @jean-gui's suggestion I added an agenda to today's systeam meeting to introduce those updated error pages. Let's ask this question and see if others have any concern about this.

Reply:
> Some ideas:
> * we could restrict this info to admins only

Here, the `exception.file` is set to be included the body of the email so it doesn't make sense to restrict it to admins only (at least if admin is systeam only) as they will not use that link.
Note that we do display `exception.file` to admins on the page itself.

> * we could only display the filename and not the full path.

Yes, we could remove the base of the file path but in order to avoid hardcoding this in the template itself, we would need another config parameter

Reply:
Conveniently, there's a filter just for that: https://symfony.com/doc/current/reference/twig_reference.html#file-relative